File: C#高级编程(第7版)源代码/502259 Code CH53/FSharp/GenericsDemo.fs
namespace Wrox.ProCSharp.Languages
open System
open System.Collections.Generic

type GenericDemo() =
    static let intList =
        new List<int>()
    static member UseGenerics() =
        intList.Add(1)
        intList.Add(2)
        intList.Add(3)

type MyGeneric<'a> when 'a :> IComparable<'a>() as this =
    let list = new List<'a>()
    member this.Add(item : 'a) = list.Add(item)
    member this.Sort() = list.Sort()
    
 

File: C#高级编程(第7版)源代码/502259 Code CH53/CSharp/GenericsDemo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp
{
    public static class GenericsDemo
    {
        public static void UseGenerics()
        {
            List<int> intList = new List<int>();
            intList.Add(1);
            intList.Add(2);
            intList.Add(3);
        }
    }

    

    public class MyGeneric<T>
        where T : IComparable<T>
    {
        private List<T> list = new List<T>();

        public void Add(T item)
        {
            list.Add(item);
        }

        public void Sort()
        {
            list.Sort();
        }
    }
}
